Feature/add redux toolkit

IgorAvshtol opened the pull request:
#What has been added:
-Add redux-toolkit;

IgorAvshtol committed:
feat: add redux-toolkit;
delete: swr package;

IgorAvshtol committed:
fix: tagsSlice;

sergeyemirzakov commented on src/components/Article/Comments.tsx:
```diff
@@ -45,33 +45,39 @@ export function Comments({ setIsOpen }: ICommentsProps) {
+                />
+              );
+            })
+          )}
```
1) Try not to use ternary in markdown. Use && pattern. I think it's more readable.

sergeyemirzakov commented on src/components/Article/Comments.tsx:
```diff
@@ -1,39 +1,39 @@
   };
+
+  useEffect(() => {
+    slug && dispatch(getComments(slug));
```
Do you need a "slug" here?

sergeyemirzakov commented on src/components/Article/Comment.tsx:
```diff
@@ -34,7 +31,7 @@ export function Comment({ author, createdAt, body, id }: ICommentProps) {
           </div>
         </div>
-        {currentUser === author.username && (
+        {user?.username === author.username && (
```
I think the "author" can also be null or undefined

sergeyemirzakov commented on src/components/Article/Comments.tsx:
```diff
@@ -1,39 +1,39 @@
-    });
-    await mutate(`/articles/${slug}/comments`);
+  const onClickSendButton = () => {
+    slug && dispatch(createComment({ slug, comment: commentText }));
```
Do you need a "slug" here?

sergeyemirzakov commented on src/store/auth/authThunk.ts:
```diff
@@ -0,0 +1,48 @@
+import { setUserFromLocalStorage } from 'services/localStorage/localStorage';
+import { IRegisterData } from 'interfaces';
+
+export const signin = createAsyncThunk(
```
rename singin to singIn

sergeyemirzakov commented on src/store/auth/authThunk.ts:
```diff
@@ -0,0 +1,48 @@
+  }
+);
+
+export const signup = createAsyncThunk(
```
rename signup to signUp

IgorAvshtol committed:
fix: codestyle in articlesThunk.ts;

IgorAvshtol committed:
fix: -changed ternary expressions on && pattern;
     -rename signin and signup thunks;

IgorAvshtol committed:
feat: -add edit ui in Header article page;

IgorAvshtol committed:
feat: add ability update and delete article;

IgorAvshtol committed:
fix: uncommented routes in App;

sergeyemirzakov commented on src/store/auth/authThunk.ts:
```diff
@@ -0,0 +1,48 @@
+    try {
+      const response = await instance.post(`users/login`, user);
+      await setUserFromLocalStorage(response.data);
+      return response.data;
```
Try to get all response. Not just the data. Because sometimes you may need "status" property to check your answer

sergeyemirzakov commented on src/components/Article/AddArticleForm/Form.tsx:
```diff
@@ -19,18 +17,31 @@ type INewArticleData = FieldValues & {
+  } = useForm<INewArticleData>({
+    mode: 'all',
+    defaultValues: {
+      title: slug && currentArticle?.title,
+      description: slug && currentArticle?.description,
+      body: slug && currentArticle?.body,
+      tagList: slug ? currentArticle?.tagList : [],
```
Why are you using "slug" here?

IgorAvshtol committed:
feat: add Success component;
fix: added the ability to get a full response in thunks;

IgorAvshtol committed:
feat: testing slices, rename Success on Popup;

IgorAvshtol committed:
Update articlesThunk.ts

IgorAvshtol committed:
feat: add Lists.tsx;

IgorAvshtol committed:
feat: add opportunity to like/dislike;
fix: ui;

IgorAvshtol committed:
fix: delete unused useeffect;

sergeyemirzakov commented on src/components/Article/Navbar.tsx:
```diff
@@ -1,22 +1,21 @@
-      toggleSignInModal();
+    if (!user) {
+      dispatch(isSignInModalOpen());
       return;
```
What do you try to return here?

sergeyemirzakov commented on src/components/Modal.tsx:
```diff
@@ -1,25 +1,24 @@
   const initRef = useRef(null);
   const onCloseModal = () => {
-    onClose();
+    signInModalOpen && dispatch(isSignInModalOpen());
+    signUpModalOpen && dispatch(isSignUpModalOpen());
     dispatch(entryIsSuccess());
   };
```
Isn't it possible to just call functions? Why do you need this check?

IgorAvshtol committed:
feat: -add ReadingLists.tsx;
      -add filter for posts(all/my);
      -add @tailwindcss/line-clamp package;
fix: ui bugs;

IgorAvshtol committed:
feat: -add Profile page;
fix: ui bugs and Tabs component;

IgorAvshtol committed:
feat: add tabsSlice;
remake Tabs.tsx;

IgorAvshtol committed:
fix: update links array;

IgorAvshtol committed:
fix: update links array;

IgorAvshtol committed:
fix: ui;
feat: add selectedTag AC, ArticleByTag;

IgorAvshtol committed:
fix: delete props for Sidebar ib ProfileContainer.tsx;

IgorAvshtol committed:
fix: code style, UI bugs;
feat: add ability update userdata;

IgorAvshtol committed:
fix: UI bugs;
feat: add the ability to save notes;

IgorAvshtol committed:
fix: code style and method map return values;

IgorAvshtol committed:
Merge pull request #25 from IgorAvshtol/feat/Add_Setting_page

fix: code style, UI bugs;

IgorAvshtol committed:
Merge pull request #24 from IgorAvshtol/feat/Add_filter_by_tag

fix: ui;

IgorAvshtol committed:
Merge pull request #23 from IgorAvshtol/feat/Remake_Tabs

feat: add tabsSlice;
